File: Programming Languages/4 languages in 2 styles/OO/Javascript/Javascript OO/Interfaces/WordCounter.mjs
export class WordCounter {
    process(fileContentList){

    }
}

export class WordCounterImpl extends WordCounter {
    process (fileContentList) {
        let fileContentMap = new Map();
        for (let i = 0; i < fileContentList.length; i++) {
            let numOfTimes = fileContentMap.get(fileContentList[i]);
            if ((numOfTimes === undefined) || (numOfTimes === "NAN")) {
                fileContentMap.set(fileContentList[i], 1);
            }
            else {
                fileContentMap.set(fileContentList[i] + "", numOfTimes + 1);
            }
        }
        return fileContentMap;
    }

}

File: Programming Languages/4 languages in 2 styles/OO/Python/Python OO/WordCounter.py
class WordCounter:
    def __init__(self):
        self

class WordCounterImpl(WordCounter):


    def process(self, file_content_list):
        thisdict = {}
        for word in file_content_list:
            if word in thisdict:
                thisdict[word] = thisdict.get(word) + 1
            else:
                thisdict[word] = 1
        return thisdict